src/index.test.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import {
+    AxeResults,
+    Result,
+    RunOptions,
+    TestEngine,
+    TestEnvironment,
+    TestRunner,
+} from 'axe-core';
+import * as fs from 'fs';
+import { axeToSarif } from './index';
+import { SarifLog } from './sarif/sarifLog';
+import { Run } from './sarif/sarifv2';
+
+describe('axe-sarf-converter integration test', () => {
+    let axeResultStub: AxeResults;
+    beforeEach(() => {
+        axeResultStub = {
+            toolOptions: {} as RunOptions,
+            testEngine: {} as TestEngine,
+            testRunner: {} as TestRunner,
+            testEnvironment: {} as TestEnvironment,
+            url: 'https://abc.com',
+            timestamp: '2018-03-23T21:36:58.321Z',
+            passes: [] as Result[],
+            violations: [] as Result[],
+            incomplete: [] as Result[],
+            inapplicable: [] as Result[],
+        };
+    });
+    it('axe-sarif-converter is defined', () => {
+        expect(axeToSarif(axeResultStub)).toBeDefined();
+    });
+
+    it('axe-sarif-converter returns a valid sarif with blank results array in run array', () => {
+        const expected: SarifLog = {
+            version: '2.0.0',
+            runs: [
+                {
+                    files: {
+                        'https://abc.com': {
+                            mimeType: 'text/html',
+                            properties: {
+                                tags: ['target'],
+                                title: '',
+                            },
+                        },
+                    },
+                    invocations: [
+                        {
+                            endTime: '2018-03-23T21:36:58.321Z',
+                            startTime: '2018-03-23T21:36:58.321Z',
+                        },
+                    ],
+                    properties: {},
+                    resources: {
+                        rules: {},
+                    },
+                    results: [],
+                    tool: {
+                        fullName: 'axe-core',
+                        name: 'axe',
+                        properties: {
+                            downloadUri: 'https://www.deque.com/axe/',
+                        },
+                        semanticVersion: '3.2.2',
+                        version: '3.2.2',
+                    },
+                },
+            ] as Run[],
+        };
+        expect(axeToSarif(axeResultStub)).toEqual(expected);
+    });
+
+    it('generates proper sarifv2 from an AxeResults stub generated by running axe-core on a test site', () => {
```

Comment:
I think "generated pinned snapshot" is probably a better name for this test than "generates proper sarifv2" - I think "generates proper sarifv2" would be a valuable integration test, but its implementation would involve validating our output against a json schema validator, not our own snapshot